company-box-doc.el

```diff
@@ -77,11 +77,17 @@
 
 (defun company-box-doc--fetch-doc-buffer (candidate)
   "Return doc for CANDIDATE."
-  (company-box--mute-apply
-    (--> (while-no-input
-           (-some-> (company-call-backend 'doc-buffer candidate)
-             (get-buffer)))
-         (if (eq t it) nil it))))
+  (let (company-mode-hook)
+    (company-box--mute-apply
+      (--> (while-no-input
+             ;; XXX: By calling `company-call-backend' with `doc-buffer' will
+             ;; enable `company-mode' once... not sure why!
+             ;;
+             ;; Let's temporarily set `company-mode-hook' to `nil' (on top) to
+             ;; prevent other possible side effects.
+             (-some-> (company-call-backend 'doc-buffer candidate)
+               (get-buffer)))
+           (if (eq t it) nil it)))))
```

Comment:
This should resolve the conflict for https://github.com/jcs-elpa/company-fuzzy/blob/822eac17b40b90748333798c3f1432d410572ff1/company-fuzzy.el#L189-L192